homeassistant/components/cast/config_flow.py

```diff
@@ -1,16 +1,25 @@
 """Config flow for Cast."""
-from pychromecast.discovery import discover_chromecasts
+import functools
+
+from pychromecast.discovery import discover_chromecasts, stop_discovery
 
 from homeassistant import config_entries
 from homeassistant.helpers import config_entry_flow
 
 from .const import DOMAIN
+from .helpers import ChromeCastZeroconf
 
 
 async def _async_has_devices(hass):
     """Return if there are devices that can be discovered."""
 
-    return await hass.async_add_executor_job(discover_chromecasts)
+    casts, browser = await hass.async_add_executor_job(
+        functools.partial(
+            discover_chromecasts, zeroconf_instance=ChromeCastZeroconf.get_zeroconf()
+        )
+    )
+    stop_discovery(browser)
```

Comment:
Is this async safe?

Reply:
Yes, it should be fine: It doesn't block and it should be fine to call from any thread.